x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/session/EsqlSessionCCSUtils.java

```diff
@@ -217,34 +210,46 @@ static void updateExecutionInfoWithClustersWithNoMatchingIndices(EsqlExecutionIn
                 }
             } else {
                 // handles local cluster (when no concrete indices requested) and skip_unavailable=true clusters
-                EsqlExecutionInfo.Cluster.Status status;
-                ShardSearchFailure failure;
+                Cluster.Status status;
+                Exception failureException;
                 if (c.equals(RemoteClusterAware.LOCAL_CLUSTER_GROUP_KEY)) {
-                    status = EsqlExecutionInfo.Cluster.Status.SUCCESSFUL;
-                    failure = null;
+                    status = Cluster.Status.SUCCESSFUL;
+                    failureException = null;
                 } else {
-                    status = EsqlExecutionInfo.Cluster.Status.SKIPPED;
-                    failure = new ShardSearchFailure(new VerificationException("Unknown index [" + indexExpression + "]"));
+                    status = Cluster.Status.SKIPPED;
+                    failureException = new VerificationException("Unknown index [" + indexExpression + "]");
                 }
-                executionInfo.swapCluster(c, (k, v) -> {
-                    var builder = new EsqlExecutionInfo.Cluster.Builder(v).setStatus(status)
-                        .setTook(new TimeValue(0))
-                        .setTotalShards(0)
-                        .setSuccessfulShards(0)
-                        .setSkippedShards(0)
-                        .setFailedShards(0);
-                    if (failure != null) {
-                        builder.setFailures(List.of(failure));
-                    }
-                    return builder.build();
-                });
+                markClusterEmptyInfo(executionInfo, c, status, failureException);
             }
         }
         if (fatalErrorMessage != null) {
             throw new VerificationException(fatalErrorMessage);
         }
     }
 
+    /**
+     * Mark cluster with an empty cluster state with the given status and potentially failure from exception.
+     */
+    public static void markClusterEmptyInfo(
```

Comment:
I don't think I know what an "empty" cluster state is. The state should be initialized to RUNNING, never empty. But you don't check for status in this method so I'm unclear what "empty" means here. Maybe you mean "empty" as in it has no data? But this only records metadata, so that's also a confusing use.

Reply:
Yeah, naming is hard. There's clearly a pattern of what we're doing here - setting all values to 0, etc. - but I am not sure how to call it. It's not necessarily failure - e.g. for some cases on local it isn't - but for remotes is usually is. If you have a better idea what to call it please suggest.